sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -437,67 +437,77 @@ public Table getTable(TableReference tableRef) throws InterruptedException, IOEx
         (int) TimeUnit.MINUTES.toMillis(5);
 
     /**
+     * {@inheritDoc}
+     *
      * Tries to create the BigQuery table.
      * If a table with the same name already exists in the dataset, the table
      * creation fails.  In such a case,
      * the existing table doesn't necessarily have the same schema as specified
      * by the parameter.
      *
-     * @param schema Schema of the new BigQuery table.
      * @throws IOException if other error than already existing table occurs.
      */
-    @Nullable
     @Override
-    public void createTable(TableReference ref, TableSchema schema) throws IOException {
-      LOG.info("Trying to create BigQuery table: {}", BigQueryIO.toTableSpec(ref));
+    public void createTable(Table table) throws IOException, InterruptedException {
+      LOG.info("Trying to create BigQuery table: {}",
+          BigQueryIO.toTableSpec(table.getTableReference()));
       BackOff backoff =
               new ExponentialBackOff.Builder()
                       .setMaxElapsedTimeMillis(RETRY_CREATE_TABLE_DURATION_MILLIS)
                       .build();
 
-      Table table = new Table().setTableReference(ref).setSchema(schema);
-      tryCreateTable(table, ref.getProjectId(), ref.getDatasetId(), backoff,
-          Sleeper.DEFAULT);
+      tryCreateTable(table, backoff, Sleeper.DEFAULT);
     }
 
     @VisibleForTesting
-    @Nullable
-    Table tryCreateTable(
-            Table table, String projectId, String datasetId, BackOff backoff, Sleeper sleeper)
-            throws IOException {
-      boolean retry = false;
-      while (true) {
+    void tryCreateTable(Table table, BackOff backoff, Sleeper sleeper)
+        throws IOException, InterruptedException {
+      Exception lastException;
+      int retryCount = 0;
+      boolean logRetryMessage = true;
+      do {
         try {
-          return client.tables().insert(projectId, datasetId, table).execute();
+          client.tables().insert(
+              table.getTableReference().getProjectId(),
+              table.getTableReference().getDatasetId(),
+              table).execute();
+          return; // SUCCEEDED
         } catch (IOException e) {
-          ApiErrorExtractor extractor = new ApiErrorExtractor();
-          if (extractor.itemAlreadyExists(e)) {
-            // The table already exists, nothing to return.
-            return null;
-          } else if (extractor.rateLimited(e)) {
```

Comment:
We should not be changing functionality in this PR. This PR is purely to add tests, not to fix functionality. Any functional changes should be in a different PR.